dask/dataframe/io/csv.py

```diff
@@ -508,6 +511,17 @@ def read_pandas(
     else:
         path_converter = None
 
+    # If compression is "infer", inspect the (first) path suffix and
+    # set the proper compression option if the suffix is recongnized.
+    if compression == "infer":
+        # Translate the input urlpath to a simple path list
+        if not isinstance(urlpath, (str, list, tuple, os.PathLike)):
+            raise TypeError("Path should be a string, os.PathLike, list or tuple")
```

Comment:
This message isn't quite right.
*IF* we allow other types at all (does file-like work? I thought not), then the message would be "compression='infer' only makes sense when passing a path or set of paths". Or something. I'm not sure we need to have the check at all.

Reply:
Okay - I think I agree that the instance check is unnecessary.  Removed it